rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/ImmutableEmptyCache.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2019. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.bookkeeper;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheStats;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.Nullable;
+
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutionException;
+
+public class ImmutableEmptyCache<K, V> implements Cache<K, V>
+{
+  public static final CacheStats STATS = new CacheStats(
+          0,
+          0,
+          0,
+          0,
+          0,
+          0);
+
+  @Nullable
+  @Override
+  public V getIfPresent(Object key)
+  {
+    return null;
+  }
+
+  @Override
+  public V get(K key, Callable<? extends V> loader)
+          throws ExecutionException
+  {
+    return null;
```

Comment:
Throwing exception explicitly would make it easier to differentiate a case of response coming from valid Cache vs this ImmutableEmptyCache.

Reply:
There isn't. The cache objects are used only in the apis which can be invoked locally during read of a file. If we add something of that sorts in future then that will fail but it wont also be right to return empty result unless expected, so there too it is good to fail loudly if this mode is missed in those kind of changes.

Reply:
Updated!